__tests__/tasks/task-dev-dependency.test.js

```diff
@@ -0,0 +1,203 @@
+const { users } = require('../../mock-data/users');
+
+describe('Dev Feature Flag Tests', () => {
+  let browser;
+  let page;
+
+  jest.setTimeout(60000);
+
+  beforeAll(async () => {
+    browser = await puppeteer.launch({
+      headless: 'new',
+      ignoreHTTPSErrors: true,
+      args: ['--incognito', '--disable-web-security'],
+      devtools: false,
+    });
+  });
+
+  beforeEach(async () => {
+    page = await browser.newPage();
+    await page.setRequestInterception(true);
+
+    // Mock API responses
+    page.on('request', (interceptedRequest) => {
+      const url = interceptedRequest.url();
+
+      if (url === `${API_BASE_URL}/levels`) {
+        interceptedRequest.respond({
+          status: 200,
+          contentType: 'application/json',
+          headers: {
+            'Access-Control-Allow-Origin': '*',
+            'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
+            'Access-Control-Allow-Headers': 'Content-Type, Authorization',
+          },
+          body: JSON.stringify(levels),
+        });
+      } else if (url === `${API_BASE_URL}/users`) {
+        interceptedRequest.respond({
+          status: 200,
+          contentType: 'application/json',
+          headers: {
+            'Access-Control-Allow-Origin': '*',
+            'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
+            'Access-Control-Allow-Headers': 'Content-Type, Authorization',
+          },
+          body: JSON.stringify(users),
+        });
+      } else if (url === `${API_BASE_URL}/tags`) {
+        interceptedRequest.respond({
+          status: 200,
+          contentType: 'application/json',
+          headers: {
+            'Access-Control-Allow-Origin': '*',
+            'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
+            'Access-Control-Allow-Headers': 'Content-Type, Authorization',
+          },
+          body: JSON.stringify(tags),
+        });
+      } else if (url === `${SKILL_TREE_BACKEND_BASE_URL}/skills`) {
+        interceptedRequest.respond({
+          status: 200,
+          contentType: 'application/json',
+          headers: {
+            'Access-Control-Allow-Origin': '*',
+          },
+          body: JSON.stringify([
+            { id: 1, name: 'JavaScript' },
+            { id: 2, name: 'React' },
+            { id: 3, name: 'Node.js' },
+          ]),
+        });
+      } else {
+        interceptedRequest.continue();
+      }
+    });
+  });
+
+  afterAll(async () => {
+    await browser.close();
+  });
+
+  describe('With dev=true', () => {
+    beforeEach(async () => {
+      await page.goto('http://localhost:8000/task?dev=true');
+      await page.waitForNetworkIdle();
+    });
+
+    it('should hide feature URL field', async () => {
+      const featureUrlField = await page.$eval(
+        '#featureUrl',
+        (el) => window.getComputedStyle(el.closest('.inputBox')).display,
+      );
```

Comment:
These fields and tests has to be removed later, i can use data-testId for this though, isnt it fine? since this has anyways later to be removed. let me know if i still should use data-testId which by the way i have used for skills field which is new addition

Reply:
Because those fields has to be removed... right now they are under feature flag (you can read more in issue description), creating a seperate file is okay

Reply:
who told you to create separate test file for these changes? Why these tests can't be written in task-dependency.test.js file?

Reply:
No one told me, What's the issue though? It can be written in same file as well

Reply:
if we can write tests in the already present file than we don't need to create a separate file for it.

Reply:
I see the concern i have reviewed your PR, if your PR get merged first i will resolve it since you are exporting base_url from mock_data/constants or vice versa. about separate file concern if you still insisit on making changes in same file... let me know i will do it